src/private.h

```diff
@@ -113,6 +113,19 @@ struct _gpuarray_blas_ops {
   int (*setup)(gpucontext *ctx);
   void (*teardown)(gpucontext *ctx);
   const char *(*error)(gpucontext *ctx);
+
+  int (*hdot)( size_t N,
+    gpudata *X, size_t offX, int incX,
+    gpudata *Y, size_t offY, int incY,
```

Comment:
Don't use int, use `ssize_t` for a signed type.  We don't want to limit the size of arrays at this level.